misc/config_tools/static_allocators/s5_vuart.py

```diff
@@ -49,23 +51,24 @@ def get_console_vuart_port(scenario_etree, vm_name):
 
 def alloc_free_port(scenario_etree, load_order, vm_name):
     port_list = scenario_etree.xpath(f"//endpoint[vm_name = '{vm_name}']/io_port/text()")
+    vm_id = int(get_node(f"//vm[name = '{vm_name}']/@id", scenario_etree))
     console_port = get_console_vuart_port(scenario_etree, vm_name)
     if console_port is not None:
         port_list.append(console_port.replace("U", ""))
+    
+    tmp_list = []
+    for port in port_list:
+        tmp_list.append(int(port, 16))
 
     if load_order == "SERVICE_VM":
-        tmp_list = []
-        for port in port_list:
-            tmp_list.append(int(port, 16))
-
-        global service_port_list
-        service_port_list = list(set(service_port_list) - set(tmp_list))
-        service_port_list.sort()
-        port = hex(service_port_list[0])
-        service_port_list.remove(service_port_list[0])
-        return str(port).upper()
-    else:
-        return "0x2F8"
+        vm_id = int(get_node(f"//vm[load_order = 'SERVICE_VM']/@id", scenario_etree))
```

Comment:
Do we still need this step? I think VM names are already unique can identify any VM regardless of its type.

Reply:
Yes, this step is necessary for Service VM to obtain vm_id